### ISSUE TITLE
S2583 FP: Invocation argument constraints should be independent

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#3288]
### Description

Reported on [this community thread](https://community.sonarsource.com/t/c-s2583-false-positive-on-null-check-if-another-type-check-is-earlier-in-the-same-function-call/22253).

### Repro steps

```C#
public static void DoSomething(object value)
    => DoSomething(
        value is null ? 1 : 2,
        value is bool b ? 3 : 4,
        value is null ? 5 : 6); // FP

private static void DoSomething(int a, int b, int c)
{
}
```
Issue is reported on last line. Removing any of previous checks (`is null` or `is bool`) removes the issue.

**Comment by mary-georgiou-sonarsource** [comment]
Fixed by https://github.com/SonarSource/sonar-dotnet/pull/7750
